kie-server-parent/kie-server-tests/kie-server-integ-tests-optaplanner/src/test/java/org/kie/server/integrationtests/optaplanner/OptaplannerIntegrationTest.java

```diff
@@ -615,6 +628,42 @@ public void testTerminateEarly() throws Exception {
                                    SOLVER_1_ID);
     }
 
+    protected WebTarget newRequest(String uriString) {
+
+        ResteasyClient httpClient = new ResteasyClientBuilder()
+                    .connectTimeout(10, TimeUnit.SECONDS)
+                    .readTimeout(10, TimeUnit.SECONDS)
+                    .build();
+
+        WebTarget webTarget = httpClient.target(uriString);
+        webTarget.register(new BasicAuthentication(TestConfig.getUsername(), TestConfig.getPassword()));
+        return webTarget;
+    }
+
+    private void checkPrometheusEndpointResponse() {
```

Comment:
Ok @rsynek it makes sense what you suggest. I'll add a test that specifically verifies the metric values taken from Prometheus. I'll try to make it just as a method but in case we need to make a separate class test - I'll do that. And yes, if Optaplanner is disabled, no OptaPlanner metrics are published. Thanks for review. I'll ping you with an update soon